crates/core/component/ibc/src/component/msg_handler/timeout.rs

```diff
@@ -83,7 +83,7 @@ impl MsgHandler for MsgTimeout {
 
         if channel.ordering == ChannelOrder::Ordered {
             // ordered channel: check that packet has not been received
-            if self.next_seq_recv_on_b != self.packet.sequence {
+            if self.next_seq_recv_on_b > self.packet.sequence {
```

Comment:
If you get a timeout with a packet sequence in the future you should still act on it, right? (which this code does)

Reply:
My understanding is that we should be acting on any packet that occured in the past and that has not been received yet, which is what ibc-go does and what this change is attempting to do.